manifests/plugin/column_view.pp

```diff
@@ -0,0 +1,26 @@
+# = Foreman Column View
+#
+# This class installs the column view plugin and optionally manages the configuration file
+#
+# === Parameters:
+#
+# $columns:: an hash of columns to add to the configuration
+#
+class foreman::plugin::column_view (
+  Optional[Hash[String,Hash[Enum['title','after','content','title','conditional','eval_content','view','width','custom_method'],String,3,9]]] $columns = undef
```

Comment:
I'd suggest looking at https://github.com/camptocamp/puppet-systemd/blob/master/types/servicelimits.pp since that results in an easier way. Another thing to consider is handling an empty hash instead of undef as default.